admin/bashrc_for_gmt

```diff
@@ -0,0 +1,97 @@
+# (dbuild directory), a release build (rbuild directory),
+# and XCode debugging (xbuild directory).
+#
+# Include in your .bashrc via source <path>/bashrc_for_gmt
+#
+# Examples:
+# Type "bnd" to pull the latest main, configure with cmake, and build
+# with debug settings, or "bnx" for the Xcode setup
+# Type "crb" to run all tests with the release build
+# Type "gmtfind 'pattern'" to find all files where 'pattern' occurs
+# Type "dlog" to open the latest test log
+# Type "vgd" to open all the failed test PNGs for build
+#
+# You must set these parameter to reflect your setup
+builder=ninja	# If ninja is not installed (please do), set to make
+Bname="Ninja"	# If ninja is not installed (please do), set to "Unix Makefiles"
+pngview=open	# Program that can open and display PNG files
+pdfview=open	# Program that can open and display PDF files
+ncores=4	# Number of cores for running tests
+MATLAB=/Applications/MATLAB_R2019a.app	# Path to the Matlab app
+REPO_DIR=<path to the local GMT repository>
+DATA_DIR=<path to directory containing GSHHG and DCW>
+EDITOR=subl		# Program that can open and display text files
+#--------------------------------------------------------------
+
+# These are needed to run admin/build-gmt-release.sh
+export GMT_GSHHG_SOURCE=${DATA_DIR}/gshhg-gmt-2.3.7
+export GMT_DCW_SOURCE=${DATA_DIR}/dcw-gmt-1.1.4
+
+# To run matlab without the GUI:
+alias matlab='$MATLAB/bin/matlab -nojvm'
+
+# function for processing find output and return list of unique files.
+function unique {
+	awk -F: '{print $1}' | sort -u
+}
+
+# List source, docs, scripts and txt files where the argument to gmtfind appears as is in the file
+# e.g, gmtfind "Grid increment is"
+function gmtfind {
+	find . -name '*.[ch]' -exec egrep -H "$*" {} \; | unique ;
+	find . -name '*.in]'  -exec egrep -H "$*" {} \; | unique ;
+	find . -name '*.txt]' -exec egrep -H "$*" {} \; | unique ;
+	find . -name '*.rst*' -exec egrep -H "$*" {} \; | unique ;
+	find . -name '*.sh'   -exec egrep -H "$*" {} \; | unique ;
+	find . -name '*.bat'  -exec egrep -H "$*" {} \; | unique ;
+}
+
+# Various GMT building aliases
+
+# To run cmake for GMT for various setups
+alias cmakegmtd='cmake -DCMAKE_INSTALL_PREFIX=gmt6 -DCMAKE_BUILD_TYPE=Debug ..' # Configure cmake for debug
+alias cmakegmtr='cmake -DCMAKE_INSTALL_PREFIX=gmt6 -DCMAKE_BUILD_TYPE=Release ..' # Configure cmake for release
+alias cmakegmtx='cmake -DCMAKE_INSTALL_PREFIX=gmt6 -DCMAKE_BUILD_TYPE=Debug -G Xcode ..' # Configure cmake for Xcode debug
+
+gmt6='cd ${REPO_DIR}/src'	# Cd to the GMT source directory
+alias gtop='cd ${REPO_DIR}'	# Cd to the top of the dev tree
+alias dlog='pushd .; gtop; ${EDITOR} dbuild/Testing/Temporary/LastTest.log; popd' # Open the dbuild check error log
+alias rlog='pushd .; gtop; ${EDITOR} rbuild/Testing/Temporary/LastTest.log; popd' # Open the rbuild check error log
+alias bdoc='pushd .; gmt6; cd ../dbuild; ${builder} -j${ncores} docs_html install; popd'
```

Comment:
@PaulWessel, can you explain why gmt is called here? I am confused about its purpose before building the docs

Reply:
Or maybe should be gtop?

Reply:
Yes, sorry, I have a gmt6 alias which cd's into src.  I think gtop; cd dbuild; is what is needed.

Reply:
Ah,  I see now thanks. It will work with gmt6, just missing an ```alias``` command I think. I can push a fix and a couple additions that I have found useful.

Reply:
Sounds good.